qdrant-landing/content/articles/serverless.md

```diff
@@ -0,0 +1,290 @@
+  Downloaded libc v0.2.137
+# [..] output omitted for brevity
+    Finished release [optimized] target(s) in 1m 27s
+$ # Upload the function
+$ aws lambda create-function --function-name $LAMBDA_FUNCTION_NAME \
+    --handler bootstrap \
+    --architectures arm64 \
+    --zip-file fileb://./target/lambda/page-search/bootstrap.zip \
+    --runtime provided.al2 \
+    --region $LAMBDA_REGION \
+    --role $LAMBDA_ROLE \
+    --tracing-config Mode=Active
+$ # Add the function URL
+$ aws lambda add-permission \
+    --function-name $LAMBDA_FUNCTION_NAME \
+    --action lambda:InvokeFunctionUrl \
+    --principal "*" \
+    --function-url-auth-type "NONE" \
+    --region $LAMBDA_REGION \
+    --statement-id url
+$ # Here for simplicity unauthenticated URL access. Beware!
+$ aws lambda create-function-url-config \
+    --function-name $LAMBDA_FUNCTION_NAME \
+    --region $LAMBDA_REGION \
+    --cors "AllowOrigins=*,AllowMethods=*,AllowHeaders=*" \
+    --auth-type NONE
+```
+
+Now you can go to your *Function Overview* and click on the Function URL. This should show something like the following:
+
+```text
+Hello, lambda!
+```
+
+Congratulations! You have set up a lambda function in Rust. On to the next ingredient:
+
+## Embedding
+
+Most providers supply a simple https GET or POST interface we can use. Most will have an API key, which you have to supply in an authentication header. Let's use OpenAI as an example, which is one of the more complex APIs. We need to extend our dependencies with `reqwest` and also add `anyhow` for easier error handling:
+
+```toml
+anyhow = "1.0"
+reqwest =  { version = "0.11.18", default-features = false, features = ["json", "rustls-tls"] }
+serde = "1.0"
+```
+
+Now given an API key, you can make a call to get the embedding vector:
+
+```rust
+use anyhow::Result;
+use serde::Deserialize;
+use reqwest::Client;
+
+#[derive(Deserialize)]
+struct OpenaiResponse { data: OpenaiData }
+
+#[derive(Deserialize)]
+struct OpanaiData { embedding: Vec<f32> }
+
+pub async fn embed(client: &Client, text: &str, api_key: &str) -> Result<Vec<f32>> {
+    let OpenaiResponse { data: OpenaiData { embedding } = client
+        .post("https://api.openai.com/v1/embedding")
+        .header("Authorization", &format!("Bearer {api_key}"))
+        .header("Content-Type", "application/json")
+        .body(format!("{{\"input\":\"{text}\",\"model\":\"text-embedding-ada-002\"}}"))
+        .send()
+        .await?
+        .json()?;
+    Ok(embedding)
+}
+```
+
+You should note that OpenAI's ada-002 model emits 1536-dimensioned vectors.
+
+Other providers have similar interfaces. Consult our [integration docs](https://qdrant.tech/documentation/integrations) for further information. See how little code it took to get the embedding? Also with [AI Grant](https://aigrant.org/), you can apply for free credits that allow for a lot of searches before you'll have to spend a dime.
+
+While we're at it, it's a good idea to write a small test:
```

Comment:
You need to say at the beginning what the test is for. Otherwise the reader has to go through the code to figure it out.